tests/app/questionnaire/conftest.py

```diff
@@ -831,5 +1022,33 @@ def mock_schema():
 
 
 @pytest.fixture
-def placeholder_transform():
-    return PlaceholderTransforms(language="en")
+def placeholder_transform(schema_placeholder_renderer):
+    schema, renderer = schema_placeholder_renderer
+    return PlaceholderTransforms(language="en", schema=schema, renderer=renderer)
+
+
+@pytest.fixture
+def schema_placeholder_renderer(section_with_labels):
+    schema = QuestionnaireSchema(section_with_labels)
+    answer_id = AnswerStore(
+        [
+            {"answer_id": "mandatory-radio-label-answer-1", "value": "Head"},
+            {"answer_id": "mandatory-radio-label-answer-2", "value": "label"},
+        ]
+    )
+    renderer = PlaceholderRenderer(
+        language="en",
+        answer_store=answer_id,
+        list_store=ListStore(),
+        metadata=ImmutableDict({}),
+        response_metadata={},
+        schema=schema,
+    )
+    return schema, renderer
```

Comment:
We should make the schema its own fixture. I think it's confusing to return unrelated things from one fixture

Reply:
To be honest, I think this should be done directly in the test it is used in. It is being set up for that specific test. I think some duplication in test placeholder transform is fine, as at the moment, it is hard to follow as the main set-up for the test is done in the fixture instead of the test body.

Reply:
They are related as this renderer and schema is specific to placeholder test. However, rest of the tests now use `mock_schema` and `mock_renderer`. fixtures Also as suggested , Placeholder is instantiated directly for labels tests